docs/eeps/draft-maven-style-plugins.md

```diff
@@ -0,0 +1,246 @@
+
+### Configuration for Maven-style plugins through Embulk System Properties
+
+The other way to configure with Maven-style plugins is Embulk System Properties. See [EEP-4](./eep-0004.md) for Embulk System Properties.
+
+Once a user sets an Embulk System Property `plugins.<category>.<type>` in the format of `maven:<group>:<name>:<version>`, the same YAML configuration as RubyGems-style keeps working. The following examples configure the same with the above to run with the S3 Input Plugin of the Maven-style plugin artifact `org.embulk:embulk-input-s3:0.5.3`.
+
+```
+# embulk.properties
+plugins.input.s3=maven:org.embulk:s3:0.5.3
+```
+
+```
+# maven-with-properties.yaml
+in:
+  type: s3
+  bucket: ...
+  path_prefix: ...
+```
+
+The `artifactId` is not fully specified like `embulk-input-s3` there like the inline configuration explained above.
+
+The latter, with Embulk System Properties, is expected to be less confusing for users.
+
+### Type conflicts
+
+The older RubyGems-style plugins can be installed along with the newer Maven-style plugins. It means they conflict. When a user specifies `type` of a plugin that has a conflict between the RubyGems-style and the Maven-style, the priority is determined as follows.
+
+* The Maven-style plugin is prioritized if the corresponding Embulk System Property `plugins.<category>.<type>` is configured, and the Maven artifact with the specified `groupId` and `version` is installed.
+* Otherwise, the RubyGems-style plugin is chosen by definition.
+
+Main class and Manifest
+------------------------
+
+Once the plugin's installed JAR artifact is determined from the user's YAML configuration, its dependencies are identified from `pom.xml` of the plugin artifact. Only the direct dependencies of the plugin artifact are included, as explained in "Intended limitation for stable dependency resolution".
+
+These JAR files are the "classpath" to be loaded for the Maven-style plugin. Embulk can load these classes now, but Embulk still needs to identify the "main class" as an Embulk plugin, which implements Embulk SPI's `*Plugin` interface, such as `org.embulk.spi.InputPlugin`, `org.embulk.spi.FilterPlugin`, and else.
+
+The main class is to be declared in `META-INF/MANIFEST.MF`, [the manifest file](https://docs.oracle.com/javase/tutorial/deployment/jar/manifestindex.html) of the plugin JAR file. The manifest file of an Embulk plugin's JAR file must include the following attributes, `Embulk-Plugin-Spi-Version`, `Embulk-Plugin-Category`, `Embulk-Plugin-Main-Class`, and `Embulk-Plugin-Type`.
+
+* `Embulk-Plugin-Spi-Version` must be `0` as of Embulk v0.11 or earlier.
+* `Embulk-Plugin-Category` is the category of the plugin, which is one of `input`, `decoder`, `parser`, `filter`, `formatter`, `encoder`, `output`, or `guess`.
```

Comment:
(I'm not familiar `pom.xml` format.)

A few days ago, @dmikurube and I talked about `embulk-parser-csv` packaging. [Twitter](https://twitter.com/dmikurube/status/1666743761295245313?s=20)
We will plan a parser and a guess package into a single jar file.
In this case, Does this attribute care about that?
(I'm wondering **one of** `parser`, .. or `guess` parts).

In those case, Does A Plugin developer creates two manifests?

Reply:
@dmikurube This part in this EEP explains only about each JAR package == each Maven artifact. `pom.xml` is per Maven artifact.

On the other hand, what we talked on `embulk-parser-csv` was not about "packaging" nor each artifact. I was talking only about the Git source code repository, not about the release package.

It intended about having two (or more) plugin source trees in the same Git repository like : 

```
embulk-parser-guess.git
    + embulk-parser-csv/
    | + build.gradle
    | + src/
    | + ...
    + embulk-guess-csv/
      + build.gradle
      + src/
      + ...
```

[`embulk-input-jdbc`](https://github.com/embulk/embulk-input-jdbc) is a similar example doing the same.

Reply:
Ah! I got it. Thanks